distrobox-enter

```diff
@@ -216,11 +216,15 @@ generate_command() {
 
 	# Entering container using our user and workdir.
 	# Start container from working directory. Else default to home. Else do /.
+	# Since we are entering from host, drop at workdir through '/run/host'
+	# which represents host's root inside container. Any directory on host
+	# even if not explicitly mounted is bound to exist under /run/host.
 	# pass distrobox-enter path, it will be used in the distrobox-export tool.
 	workdir="$(echo "${PWD:-${HOME:-"/"}}" | sed -e 's/"/\\\"/g')"
+	workdir="/run/host/${workdir}"
```

Comment:
Let's continue discussing this solution:

```sh
workdir="$(echo "${PWD:-${HOME:-"/"}}" | sed -e 's/"/\\\"/g')"
if [ -n "${workdir##*${HOME}*}" ]; then
                 workdir="/run/host${workdir}"
fi
```

Reply:
I've been testing this soln. And it does make the exclusion explicit for Host's $HOME subdirectories. 👍
But I must note it does not have any effect on custom homes' subdirectories (that are not under Host $HOME).

See following commit for tests output.

Reply:
## Test Setup

```
[ennec-e@fedora distrobox]$ git diff
diff --git a/distrobox-enter b/distrobox-enter
index b08451f..e5628c7 100755
--- a/distrobox-enter
+++ b/distrobox-enter
@@ -218,6 +218,9 @@ generate_command() {
        # Start container from working directory. Else default to home. Else do /.
        # pass distrobox-enter path, it will be used in the distrobox-export tool.
        workdir="$(echo "${PWD:-${HOME:-"/"}}" | sed -e 's/"/\\\"/g')"
+       if [ -n "${workdir##*${HOME}*}" ]; then
+               workdir="/run/host${workdir}"
+       fi
        result_command="${result_command}
                --workdir=\"${workdir}\"
                --env=\"DISTROBOX_ENTER_PATH=${distrobox_enter_path}"\"
[ennec-e@fedora distrobox]$ distrobox-create -i ubuntu:latest -H /tmp/test-exception-custom-home
```
Also install tmux in guest.

## Test Scenarios

### Simple enter from host $HOME subdir

```
[ennec-e@fedora distrobox]$ distrobox-enter ubuntu-latest -- bash -l
ennec-e@ubuntu-latest:/home/ennec-e/Code/distrobox$ 
```
👍

### Tmux enter from host $HOME subdir
```
[ennec-e@fedora distrobox]$ distrobox-enter ubuntu-latest -- tmux
---
ennec-e@ubuntu-latest:/home/ennec-e/Code/distrobox$ 
---
```
👍

### Tmux enter from custom home subdir
```
[ennec-e@fedora test-exception-custom-home]$ distrobox-enter ubuntu-latest -- tmux
---
ennec-e@ubuntu-latest:/run/host/tmp/test-exception-custom-home$ 
---
```
❗😵‍💫

### Simple enter from custom home subdir
 
```
[ennec-e@fedora test-exception-custom-home]$ distrobox-enter ubuntu-latest -- bash -l
ennec-e@ubuntu-latest:~$ 
```
👍

Reply:
The fourth case works as usual because of bash magic as described in a [previous comment]. (Same applies to case 1)
The solution presented does explicitly exclude, _host's_ $HOME so it does work for case 1 and 2. 

At enter time distrobox-enter does not know about _guest's_ custom $HOME dir, so it may not be excludeable from there. 
The solution for 3rd case I think would have to run somewhere inside the guest or from a script that is aware of custom home value. 

I personally don't really mind at all. case 3 might be some sort of an edge case anyway (custom home _outside_ Host home + custom shell). And perhaps can be left documented.

Otherwise this looks good enough : ) It makes this patch consistent with toolbox behaviour. So great!

[previous comment]:  https://github.com/89luca89/distrobox/pull/186#issuecomment-1050421175

Reply:
Technically case 3 (custom home) is not thought to enter in the custom home by default, but still preserve the workdir you are.  It is there just to have a custom home, so I think it's an edge case that we can work out later :+1:

Reply:
Have added a commit with the requested change made. Please review.